File: Blind75/1-twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash_table = {}
        for i in range(len(nums)):
            diff = target - nums[i] # check if diff is in hash table
            if diff in hash_table:
                return [i, hash_table[diff]] # if so, return indices
            else:
                hash_table[nums[i]] = i

File: Blind75/1-twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

// create a hash table
var twoSum = function(nums, target) {
    const map = new Map();
    for (let i = 0; i < nums.length; i++) {
        let diff = target - nums[i];
        if(map.has(diff)){
            return [map.get(diff), i]
        }
        map.set(nums[i], i)
    }
}
